test/op_shape_test.cpp

```diff
@@ -2325,6 +2325,218 @@ TEST_CASE(test_scalar_nelemnts)
     throws_shape(migraphx::make_op("scalar", {{"scalar_bcst_dims", {2, 3, 4, 5}}}), input);
 }
 
+TEST_CASE(test_gathernd)
+{
+    {
+        // k > r
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape is{itype, {2, 4}};
+        migraphx::shape ds{dtype, {8}};
+
+        int batch_dims(1);
+        throws_shape(migraphx::make_op("gathernd", {{"batch_dims", batch_dims}}), ds, is);
+    }
+
+    {
+        // k > r - batch_dims
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape is{itype, {2, 4}};
+        migraphx::shape ds{dtype, {2}};
+
+        int batch_dims(1);
+        throws_shape(migraphx::make_op("gathernd", {{"batch_dims", batch_dims}}), ds, is);
+    }
+
+    {
+        // batch_dims >= r
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape is{itype, {2, 1}};
+        migraphx::shape ds{dtype, {2, 5, 6, 7}};
+
+        int batch_dims(3);
+        throws_shape(migraphx::make_op("gathernd", {{"batch_dims", batch_dims}}), ds, is);
+    }
+
+    {
+        // int(q) + r - k - batch_dims - 1 = 0
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape is{itype, {2, 4}};
+        std::vector<size_t> b{2};
+        migraphx::shape ds{dtype, b};
+
+        throws_shape(migraphx::make_op("gathernd"), is, ds);
```

Comment:
Shouldn't rank `0` output be valid? that means it is a scalar value. I don't find any documentation on ONNX for that. 
++++ @CharlieL7

Reply:
To get a scalar, indices tensor must be rank 1 `q = 1`. Let's say `r = 2`, `batch_dims = 0`. Therefore `q + r - k - batch_dims - 1 = 0` goes to `1 + 2 - k - 0 - 1 = 0`. So `k = 2`. That looks like it could work. Would need to change how the code works to handle this case.

Reply:
Just add a throw for this case. I do not think we will see it often.

Reply:
I added this feature and it tests out OK.  A calculation that leads to rank 0 returns, instead, a 1 dimensional shape with a size of 1.